apps/nrm/behaviours/application-saved.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+const saveAndExitService = require('../services/save-and-exit-service');
+
+/**
+ * Save data to database
+ *
+ * @param {object} data - request object
+ *
+ * @returns {Promise}
+ */
+const saveData = async(data) => {
+  return await saveAndExitService.write(data);
+};
+
+
+/**
+ * Get Session data
+ *
+ * @param {object} req - request object
+ *
+ * @returns {object} - the session data in the request object
+ */
+const getSessionData = (req) => {
+  const sessionData = req.sessionModel.toJSON();
+
+  return sessionData;
+};
+
+/**
+ * Save application progress
+ *
+ * Get the application data from the session then persist the data
+ * within a database
+ *
+ * @param {object} req - request object
+ *
+ * @returns {Promise} - id returned from saved application
+ */
+const saveApplication = async(req) => {
+  const sessionData = getSessionData(req);
+  return await saveData(sessionData);
+};
+
+/**
+ * Check if application has been saved successfully
+ *
+ * If the 'application-id' has been set then an application has been
```

Comment:
Expand on more what this. WHY is the variable necessary.

In case the data isn't saved in the DB We need to present that to the user somehow. This will let me do that.